lib/ansible/modules/cloud/amazon/aws_acm.py

```diff
@@ -0,0 +1,403 @@
+        elif module.params['certificate_arn']:
+            module.fail_json(msg="Parameter 'certificate_arn' is only valid if parameter 'state' is specified as 'absent'")
+        elif not module.params['name_tag']:
+            module.fail_json(msg="Parameter 'name_tag' must be specified if parameter 'state' is specified as 'present'")
+        elif not module.params['private_key']:
+            module.fail_json(msg="Parameter 'private_key' must be specified if 'state' is specified as 'present'")
+    else:  # absent
+
+        # exactly one of these should be specified
+        absent_args = ['certificate_arn', 'domain_name', 'name_tag']
+        if sum([(module.params[a] is not None) for a in absent_args]) != 1:
+            for a in absent_args:
+                module.debug("%s is %s" % (a, module.params[a]))
+            module.fail_json(msg="If 'state' is specified as 'absent' then exactly one of 'name_tag', certificate_arn' or 'domain_name' must be specified")
+
+    if module.params['name_tag']:
+        tags = dict(Name=module.params['name_tag'])
+    else:
+        tags = None
+
+    region, ec2_url, aws_connect_kwargs = get_aws_connection_info(module, boto3=True)
+    client = boto3_conn(module, conn_type='client', resource='acm',
+                        region=region, endpoint=ec2_url, **aws_connect_kwargs)
+
+    # fetch the list of certificates currently in ACM
+    certificates = acm.get_certificates(client=client,
+                                        module=module,
+                                        domain_name=module.params['domain_name'],
+                                        arn=module.params['certificate_arn'],
+                                        only_tags=tags)
+
+    module.debug("Found %d corresponding certificates in ACM" % len(certificates))
+
+    if module.params['state'] == 'present':
+        if len(certificates) > 1:
+            msg = "More than one certificate with Name=%s exists in ACM in this region" % module.params['name_tag']
+            module.fail_json(msg=msg, certificates=certificates)
+        elif len(certificates) == 1:
+            # update the existing certificate
+            module.debug("Existing certificate found in ACM")
+            old_cert = certificates[0]  # existing cert in ACM
+            if ('tags' not in old_cert) or ('Name' not in old_cert['tags']) or (old_cert['tags']['Name'] != module.params['name_tag']):
+                # shouldn't happen
+                module.fail_json(msg="Internal error, unsure which certificate to update", certificate=old_cert)
+
+            if 'certificate' not in old_cert:
+                # shouldn't happen
+                module.fail_json(msg="Internal error, unsure what the existing cert in ACM is", certificate=old_cert)
+
+            # Are the existing certificate in ACM and the local certificate the same?
+            same = True
+            same &= chain_compare(old_cert['certificate'], module.params['certificate'])
```

Comment:
```suggestion
            same &= chain_compare(module, old_cert['certificate'], module.params['certificate'])
```
And the 2 below uses.